internal/exp/metrics/streams/streams.go

```diff
@@ -61,3 +65,15 @@ func (m HashMap[T]) Clear() {
 type Evictor interface {
 	Evict() identity.Stream
 }
+
+type DataPointSlice[DP DataPoint[DP]] interface {
```

Comment:
The changes in this file seemed unrelated to this pr?

Reply:
It's used in the datapoint merge function: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32794/files#diff-756513e8ec6edd51e0dc85ef3e3756b9674a2929328395fc63e806e9d052da46R113

I'd also like to migrate `processor/intervalprocessor` to use this shared definition, instead of having a duplicate interface in multiple places. https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/intervalprocessor/internal/metrics/metrics.go